pkg/build/apiserver/buildgenerator/generator.go

```diff
@@ -338,30 +355,55 @@ func (g *BuildGenerator) checkLastVersion(bc *buildv1.BuildConfig, lastVersion *
 }
 
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
-// updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
+// updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference.
+// Also updates the LastTriggerTime for the trigger that correlates to the triggeredBy parameter.
 func (g *BuildGenerator) updateImageTriggers(ctx context.Context, bc *buildv1.BuildConfig, from, triggeredBy *corev1.ObjectReference) error {
 	var requestTrigger *buildv1.ImageChangeTrigger
+	var responseTrigger *buildv1.ImageChangeTriggerStatus
+	strategyImageRef := buildutil.GetInputReference(bc.Spec.Strategy)
 	if from != nil {
-		requestTrigger = findImageChangeTrigger(bc, from)
+		requestTrigger, responseTrigger = findImageChangeTrigger(bc, from)
 	}
-	if requestTrigger != nil && triggeredBy != nil && requestTrigger.LastTriggeredImageID == triggeredBy.Name {
+	if triggeredBy != nil &&
+		//TODO we still update spec until deprecated field (drepcated in 4.8) is removed (presumably 4.9)
+		((requestTrigger != nil && requestTrigger.LastTriggeredImageID == triggeredBy.Name) ||
+			(responseTrigger != nil && responseTrigger.LastTriggeredImageID == triggeredBy.Name)) {
 		klog.V(2).Infof("Aborting imageid triggered build for BuildConfig %s/%s with imageid %s because the BuildConfig already matches this imageid", bc.Namespace, bc.Name, triggeredBy.Name)
 		return fmt.Errorf("build config %s/%s has already instantiated a build for imageid %s", bc.Namespace, bc.Name, triggeredBy.Name)
 	}
 	// Update last triggered image id for all image change triggers
+	// reset the status field in case the list of ICTs change in the spec; we'll then repopulate all of them
+	bc.Status.ImageChangeTriggers = []buildv1.ImageChangeTriggerStatus{}
 	for _, trigger := range bc.Spec.Triggers {
 		if trigger.Type != buildv1.ImageChangeBuildTriggerType {
 			continue
 		}
+		ictsFrom := trigger.ImageChange.From
+		if ictsFrom == nil {
+			ictsFrom = strategyImageRef
+		}
+		imageChangeTriggerStatus := buildv1.ImageChangeTriggerStatus{
+			From:   ictsFrom,
+			Paused: trigger.ImageChange.Paused,
+		}
 		// Use the requested image id for the trigger that caused the build, otherwise resolve to the latest
 		if triggeredBy != nil && trigger.ImageChange == requestTrigger {
+			//TODO we still update spec until deprecated field (drepcated in 4.8) is removed (presumably 4.9)
```

Comment:
nit on deprecation:
```suggestion
			//TODO we still update spec until deprecated field (deprecated in 4.8) is ignored (presumably 4.9)
```

I don't think we can actually remove the field, but we can ignore it permanently.